### PR TITLE
Explicitly call exthosttelemetry on command failure

### LINKED ISSUE #172132: Error handler not triggered when triggering an error
Testing #172057

I added this code to my extension:

```ts
vscode.window.createTextEditorDecorationType(null!);

setTimeout(() => {
		throw new Error("HELLO WORLD");
	}, 5000);

const logger = vscode.env.createTelemetryLogger({
	sendEventData: function (eventName: string, data?: Record<string, any> | undefined): void {
		console.log("sendEventData: ", eventName, data);
	},
	sendErrorData: function (error: Error, data?: Record<string, any> | undefined): void {
		console.log("sendErrorData: ", error, data);
	}
}, {});
```

But my error data handler was not triggered.

### PR DESCRIPTION
**Pull request by lramos15**
Fixes #172132

### CODE REVIEW

**Review by Tyriar — DISMISSED** [review]

**Comment by lramos15** [comment]
FYI @jrieken I removed a duplicate test and merged another one because there was warnings about them in the editor. Feel free to revert that if you don't want those changes since I was just cleaning up while in the area.

**Review by Tyriar — DISMISSED** [review]

File: src/vs/workbench/api/test/browser/extHostApiCommands.test.ts [hunk 1]
```diff
@@ -1486,26 +1490,6 @@ suite('ExtHostLanguageFeatureCommands', function () {
...
-		await rpcProtocol.sync();
-		const value = await commands.executeCommand<vscode.SelectionRange[]>('vscode.executeSelectionRangeProvider', model.uri, [new types.Position(0, 10)]);
-		assert.strictEqual(value.length, 1);
-		assert.strictEqual(value[0].range.start.line, 0);
-		assert.strictEqual(value[0].range.start.character, 10);
-		assert.strictEqual(value[0].range.end.line, 0);
-		assert.strictEqual(value[0].range.end.character, 10);
-	});
-
 	test('selectionRangeProvider on inner array always returns outer array #91852', async function () {
```
Comment (jrieken) [inline src/vs/workbench/api/test/browser/extHostApiCommands.test.ts:1494]: This is just a name duplication but a (slighty) different test (provider returns one or two elements) so we should keep both and simply rename them

File: src/vs/workbench/api/test/browser/extHostLanguageFeatures.test.ts [hunk 2]
```diff
@@ -872,11 +876,8 @@ suite('ExtHostLanguageFeatures', function () {
 		await rename(languageFeaturesService.renameProvider, model, new EditorPosition(1, 1), 'newName');
 
 		assert.deepStrictEqual(called, [true, true, true, false]);
-	});
-
-	test('Multiple RenameProviders don\'t respect all possible PrepareRename handlers, #98352', async function () {
 
-		const called = [false, false, false];
+		called = [false, false, false];
```
Comment (jrieken) [inline src/vs/workbench/api/test/browser/extHostLanguageFeatures.test.ts:880]: Better keep the test but use a different name?

**Review by jrieken — CHANGES_REQUESTED** [review]
Let's simply rename test to make them unique

**Comment by Tyriar** [comment]
Integration tests failing

**Comment by lramos15** [comment]
> Integration tests failing

Yeah waiting to hear back from @jrieken about possible thoughts. The `IExtHostTelemetry` initiates RPC messages to sync the levels across processes so now by adding it to a dependency of commands there is an RPC message that goes through when the service is instantiated that wasn't there before. It's just 1 so not really a perf issue, but we were asserting no RPC messages at all before.

**Review by jrieken — APPROVED** [review]

**Pull request merged**